src/libs/actions/Report.js

```diff
@@ -1084,6 +1086,36 @@ function deleteReportComment(reportID, reportAction) {
                 };
 
                 Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, reportActionsToMerge);
+            } else {
+                // Update last message information in the report object
+                Onyx.connect({
+                    key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+                    callback: _.once((reportActionList) => {
+                        // Find last non-empty message
+                        const lastReportAction = !_.isEmpty(reportActionList)
+                            ? _.find(Object.values(reportActionList).reverse(),
+                                ({message, actionName}) => _.last(message).html !== ''
+                                && actionName !== 'CREATED')
+                            : null;
+
+                        // Populate last message details
+                        const lastMessageDetails = {};
+                        if (lastReportAction) {
+                            // Replace line breaks with spaces and remove all HTML from the message
+                            lastMessageDetails.lastMessageText = lastReportAction.isAttachment
+                                ? '[Attachment]'
+                                : Str.stripHTML(_.last(lastReportAction.message).html.replace(/((<br[^>]*>)+)/gi, ' '));
+                            lastMessageDetails.lastMessageTimestamp = lastReportAction.timestamp;
+                            lastMessageDetails.lastActorEmail = lodashGet(lastReportAction, 'actorEmail', '');
+                        } else {
+                            lastMessageDetails.lastMessageText = '';
+                            lastMessageDetails.lastMessageTimestamp = 0;
+                            lastMessageDetails.lastActorEmail = '';
+                        }
+
+                        Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, lastMessageDetails);
```

Comment:
Please release the connection here before this statement. And remove `_.once`. Could you please try to think of your solution in an optimistic way?

Reply:
I am aware this is not the best approach. As I stated in one of the messages above, ideally we move this entire code to `updateReportActionMessage` function and have the handling for delete action next to edit action. However, until https://github.com/Expensify/Expensify.cash/issues/3524 is resolved this cannot be done for delete action.